src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/Manifest.cs

```diff
@@ -36,13 +36,20 @@ public Manifest(ManifestMetadata metadata, ICollection<ManifestFile> files)
             if (files != null)
             {
                 Files = files;
+                HasFilesNode = true;
+            }
+            else
+            {
+                HasFilesNode = false;
             }
         }
 
         public ManifestMetadata Metadata { get; }
 
         public ICollection<ManifestFile> Files { get; } = new List<ManifestFile>();
 
+        public bool HasFilesNode { get; }
```

Comment:
Making it null to indicate that there was no Files node was an option.  It would require checking for null in over a dozen places that currently assume it's not null.  I talked about it with Justin and we both liked this better.